crates/fuel-streams-core/src/outputs/subjects.rs

```diff
@@ -0,0 +1,310 @@
+use fuel_core_types::fuel_types::{Address, Bytes32, ContractId};
+use fuel_streams_macros::subject::{IntoSubject, Subject};
+
+use crate::types::*;
+
+/// Represents the NATS subject for all outputs.
+///
+/// This subject format allows for querying all outputs, optionally filtered by
+/// transaction ID and index.
+///
+/// # Examples
+///
+/// **Creating a subject for all outputs of a specific transaction:**
+///
+/// ```
+/// use fuel_streams_core::outputs::subjects::OutputsAllSubject;
+/// use fuel_streams_core::types::*;
+/// use fuel_streams_macros::subject::SubjectBuildable;
+///
+/// let subject = OutputsAllSubject::new()
+///     .with_tx_id(Some(Bytes32::zeroed()))
+///     .with_index(Some(0));
+/// assert_eq!(
+///     subject.to_string(),
+///     "outputs.0x0000000000000000000000000000000000000000000000000000000000000000.0.>"
+/// );
+/// ```
+#[derive(Subject, Debug, Clone, Default)]
+#[subject_wildcard = "outputs.>"]
+#[subject_format = "outputs.{tx_id}.{index}.>"]
```

Comment:
```suggestion
#[subject_format = "outputs.{tx_id}.{index}"]
```

Reply:
Sorry, sir, I'm not getting your points here. Explain to me why you need this `OutputsAllSubject` in the context of our system. If this is supposed to be a subject to be used inside the `fuel-streams` as a `Filter` or to be published using it in the publisher, this `>` doesn't make sense because you are literally populating `>` in the subject _(in fact, this is the key of the store key)_

As you can [see here](https://github.com/FuelLabs/data-systems/blob/main/crates/fuel-streams-macros/subject-derive/src/parse_fn.rs#L16), this `subject_format` is used to do a parse, this is why the word inside the `{}` need to be the same of the one you passed to the struct, by adding a `>` in the end, you are literally publishing the `>`, this is not used to query.

And, by populating a subject with `>` on it, you are going against what NATS recommends, which is to avoid using `>` literally in the subject.
![CleanShot 2024-09-18 at 18 14 08@2x](https://github.com/user-attachments/assets/db0c5f15-6338-42b8-afc9-58897f1805d0)

Do you know what I mean?